model/src/main/java/org/openremote/model/teltonika/State.java

```diff
@@ -0,0 +1,298 @@
+                        // If it was properly defined, then parameter ID 237 wouldn't be inaccurate, let alone to this state.
+                        // Not to mention parsing errors (from example from the UDP/Codec 8 to MQTT/Codec JSON converter, look at accelerator axes)
+                        // THE AXES VARS OVERFLOW  - if i see that TCT has value -46, if I do 65535 minus the value I am given, then it gives me the real value
+                        if (min.isPresent()) {
+                            if ( !((Double) value.get() < (Double) min.get())) {
+                                constraintValues.add(new ValueConstraint.Min((Double) min.get() * finalMultiplier));
+                            }
+                        }
+                        if(max.isPresent()){
+                            if ( !((Double) value.get() > (Double) max.get())) {
+                                constraintValues.add(new ValueConstraint.Max((Double) max.get() * finalMultiplier));
+                            }
+                        }
+
+                        ValueConstraint[] constraints = constraintValues.toArray(new ValueConstraint[0]);
+
+                        constraintsMeta.setValue(constraints);
+
+                        //TODO: Fix this, constraints for some reason are not being applied
+//                        metaMap.add(constraintsMeta);
+                    }
+                }catch (Exception e){
+                    logger.severe(parameter.propertyName + "Failed constraints");
+                    logger.severe(e.toString());
+                    throw e;
+                }
+            }
+            // Add on its label
+            try{
+                MetaItem<String> label = new MetaItem<>(MetaItemType.LABEL);
+                label.setValue(parameter.propertyName);
+                metaMap.add(label);
+            }catch (Exception e){
+                logger.severe(parameter.propertyName + "Failed label");
+                logger.severe(e.toString());
+                throw e;
+            }
+            //Use the MetaMap to create an AttributeDescriptor
+            AttributeDescriptor<?> attributeDescriptor = new AttributeDescriptor<>(parameter.propertyId.toString(), attributeType, metaMap);
+
+
+
+            //Use the AttributeDescriptor and the Value to create a new Attribute
+            Attribute<?> generatedAttribute = new Attribute(attributeDescriptor, value.get());
+            // Add it to the AttributeMap
+            attributes.addOrReplace(generatedAttribute);
+        }
+        //Special parameter definitions without being defined in the AVL Parameter List, thanks Teltonika
+
+        //latlng are the latitude-longitude coordinates, also check if it's 0,0, if it is, don't update.
+        if (reported.containsKey("latlng") && !Objects.equals(reported.get("latlng"), "0.000000,0.000000")){
+            try{
```

Comment:
unlikely but what if the device is at this actual location

Reply:
The only reason I do this is because the device will send those coordinates if it isn't able to get a satellite fix. Not to mention, this coordinate is quite far out at sea, but at that point, we can think of something else. I am only doing this because I would like the timestamps of attributes to be when the actual last data of the attribute was sent, instead of when dummy data was sent (like 0,0), for other reasons. Also, this is configurable on the device-side, meaning that the user can configure if the device will send packets if there is a satellite fix or not